CallAutomation_CallLiveTranscription/CallAutomationLiveTranscription/Program.cs

```diff
@@ -0,0 +1,400 @@
+var app = builder.Build();
+
+string helpIVRPrompt = "Welcome to the Contoso Utilities. To access your account, we need to verify your identity. Please enter your date of birth in the format DDMMYYYY using the keypad on your phone. Once we�ve validated your identity we will connect you to the next available agent. Please note this call will be recorded!";
+string addAgentPrompt = "Thank you for verifying your identity. We are now connecting you to the next available agent. Please hold the line and we will be with you shortly. Thank you for your patience.";
+string incorrectDobPrompt = "Sorry, we were unable to verify your identity based on the date of birth you entered. Please try again. Remember to enter your date of birth in the format DDMMYYYY using the keypad on your phone. Once you've entered your date of birth, press the pound key. Thank you!";
+string addParticipantFailurePrompt = "We're sorry, we were unable to connect you to an agent at this time, we will get the next available agent to call you back as soon as possible.";
+string goodbyePrompt = "Thank you for calling Contoso Utilities. We hope we were able to assist you today. Goodbye";
+string timeoutSilencePrompt = "I�m sorry, I didn�t receive any input. Please type your date of birth in the format of DDMMYYYY.";
+string goodbyeContext = "Goodbye";
+string addAgentContext = "AddAgent";
+string incorrectDobContext = "IncorrectDob";
+string addParticipantFailureContext = "FailedToAddParticipant";
+string DobRegex = "^(0[1-9]|[12][0-9]|3[01])(0[1-9]|1[012])[12][0-9]{3}$";
+bool isTrasncriptionActive = false;
+var maxTimeout = 2;
+
+string recordingId = string.Empty;
+string recordingLocation = string.Empty;
+
+app.MapPost("/api/incomingCall", async (
+    [FromBody] EventGridEvent[] eventGridEvents,
+    ILogger<Program> logger) =>
+{
+    foreach (var eventGridEvent in eventGridEvents)
+    {
+        logger.LogInformation("Call event received:{eventType}", eventGridEvent.EventType);
+
+        /* Handle system events */
+        if (eventGridEvent.TryGetSystemEventData(out object eventData))
+        {
+            /* Handle the subscription validation event. */
+            if (eventData is SubscriptionValidationEventData subscriptionValidationEventData)
+            {
+                var responseData = new SubscriptionValidationResponse
+                {
+                    ValidationResponse = subscriptionValidationEventData.ValidationCode
+                };
+                return Results.Ok(responseData);
+            }
+        }
+
+        if (eventData is AcsIncomingCallEventData incomingCallEventData)
+        {
+            var callerId = incomingCallEventData.FromCommunicationIdentifier.RawId;
+            var callbackUri = new Uri(new Uri(callbackUriHost), $"/api/callbacks/{Guid.NewGuid()}?callerId={callerId}");
+            logger.LogInformation("Incoming call - correlationId: {cor}, Callback url: {callbackuri}, transport Url: {transurl}",
```

Comment:
Nit: lets avoid using the short names like transurl, cor etc if possible. Consider using string interpolation ( $"" ) will make it cleaner i believe. I see we are using at some places it would be good to be consistent across the sample